lib/rules/no-debug.ts

```diff
@@ -107,7 +108,11 @@ export default createTestingLibraryRule<Options, MessageIds>({
           return;
         }
 
-        const isDebugUtil = helpers.isDebugUtil(callExpressionIdentifier);
+        const isDebugUtil =
+          helpers.isDebugUtil(callExpressionIdentifier) &&
+          (!isMemberExpression(node.callee) ||
+            !ASTUtils.isIdentifier(node.callee.object) ||
+            node.callee.object.name !== 'console');
```

Comment:
Can you extract this into something more explicit? This sort of chained checks were really difficult to follow when refactoring to v4. Something like this would be enough and easier to understand:

```suggestion
        const isGlobalConsoleProperty = (isMemberExpression(node.callee) &&
            ASTUtils.isIdentifier(node.callee.object) &&
            node.callee.object.name === 'console');
            
        const isDebugUtil = helpers.isDebugUtil(callExpressionIdentifier) && !isGlobalConsoleProperty;
```

Reply:
It could even be done inside `isDebugUtil` since it always applies to Testing Library `debug` util.

Reply:
I liked this idea - I applied it here https://github.com/testing-library/eslint-plugin-testing-library/pull/349/commits/eb5748f4a475954648335e648bf422067517ee6c